src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -293,6 +293,38 @@ internal static string FindBestVersionMatch(VersionRange versionRange, IEnumerab
             return matchingVersion?.ToString();
         }
 
+        private NuGetVersion ResolvePlatformReleaseChannelVersion(IList<NuGetVersion> orderedByDescBundles) => _platformReleaseChannel switch
+        {
+            ScriptConstants.StandardPlatformChannelNameUpper or ScriptConstants.ExtendedPlatformChannelNameUpper => GetStandardOrExtendedBundleVersion(orderedByDescBundles),
+            ScriptConstants.LatestPlatformChannelNameUpper or "" => GetLatestBundleVersion(orderedByDescBundles),
+            _ => HandleUnknownPlatformReleaseChannelName(orderedByDescBundles)
+        };
+
+        private NuGetVersion GetStandardOrExtendedBundleVersion(IList<NuGetVersion> orderedByDescBundlesList)
+        {
+            if (orderedByDescBundlesList.Count > 1)
+            {
+                return orderedByDescBundlesList[1]; // These channels should resolve to the version prior to latest. This list is in descending order, which makes latest [0], and prior-to-latest [1].
+            }
+
+            // keep the latest version, log a notice
+            var latest = orderedByDescBundlesList.FirstOrDefault();
+            _logger.LogInformation("Unable to apply platform release channel configuration {platformReleaseChannelName}. Only one matching bundle version is available. {latestBundleVersion} will be used", _platformReleaseChannel, latest);
```

Comment:
The usage of `FirstOrDefault()` above, specifically the _OrDefault_ part means `latest` could be `null`. Or are there other safeguards in place to ensure there is at least 1 element in the list?

Should this be a `.First()`, or do you intentionally want this log to include `latestBundleVersion: null` for investigation purposes?

Reply:
Logging `null` is intentional as having a `null` value resolved for bundle resolution is currently supported behavior (see @soninaren's [comment](https://github.com/Azure/azure-functions-host/pull/10790/#discussion_r1945589167)).